src/sched/lthread.c

```diff
@@ -461,6 +487,50 @@ void _lthread_free(struct lthread* lt)
     lt = 0;
 }
 
+int __init_utp(void *p, int set_tp)
+{
+	struct lthread_tcb_base *tcb = (struct lthread_tcb_base *)p;
+	tcb->self = p;
+	tcb->schedctx = __scheduler_self();
+	if (libc.user_tls_enabled && set_tp)
+	{
+		if (sgxlkl_enclave->mode == SW_DEBUG_MODE)
+		{
+			int r = __set_thread_area(TP_ADJ(p));
```

Comment:
As I recall, this is implemented in musl.  If so, we need to remove it.  I believe that, in software mode, this should do a host syscall to set the %fs register value.